libs/langchain/langchain/vectorstores/mongodb_atlas.py

```diff
@@ -60,6 +61,7 @@ def __init__(
         index_name: str = "default",
         text_key: str = "text",
         embedding_key: str = "embedding",
+        compatibility_mode: bool = False,
```

Comment:
My gut says we should remove this for now. I think the risk a user encounters an issue where they would need to use this feature is low. We can always add this later if we find out it's necessary for some use-case we haven't considered.

Reply:
Done!